layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -863,7 +863,11 @@ ones created by `magit' and `dired'."
 (defun spacemacs/copy-file-path ()
   "Copy and show the file path of the current buffer."
   (interactive)
-  (if-let* ((file-path (spacemacs--file-path)))
+  (if-let* ((file-path (or (spacemacs--file-path)
+                           (and (eq major-mode 'dired-mode)
+                                (condition-case nil
+                                    (dired-get-file-for-visit)
```

Comment:
If in a dired buffer and `dired-get-file-for-visit`, it should probably be an error rather than silently doing a different behavior, IMO.  Users can already press `SPC f y d` to get the current directory.

Reply:
This PR wan't the function be more "smart", so try best to copy the the path as much as possible.

Reply:
On a dired buffer, function `copy-file-path` to copy dir path still makes sense for user.

Reply:
"Smart" can be confusing instead.  I would find `SPC f y y` *sometimes* copying a file and *sometimes* copying a directory to be a very unintuitive behavior.

Reply:
An emacs user try `find-file with a dir "~/", or `(find-file "~/")`, then emacs open a dired buffer, that imply a directory is a specific file in emacs; so user try a copy file path function on a dired buffer, copying  the dir path to user still makes sense.